### PR TITLE
 Add ESLint and  Format Files

### PR DESCRIPTION
**Pull request by codenirvana**

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -1,48 +1,84 @@
...
-        var stats = fs.lstatSync(path),
-            file;
+    /**
+     * Check if string ends with the specified suffix.
+     *
+     * @param {String} str - Input string
+     * @param {String} suffix - suffix to be checked
+     * @returns {Boolean}
+     */
+    endsWith = (str, suffix) => {
```
Comment (kunagpal) [inline lib/utils.js:14]: These have to be replaced with Lodash equivalents in a separate pull request.

File: package.json [hunk 2]
```diff
@@ -42,6 +42,11 @@
     "chai": "4.1.2",
     "colors": "1.3.2",
     "editorconfig": "0.15.0",
+    "eslint": "5.5.0",
```
Comment (kunagpal) [inline package.json:45]: ~Update versions.~

File: scripts/build-docs.js [hunk 3]
```diff
@@ -19,118 +19,157 @@ var path = require('path'),
...
-function getStableVersions(versions) {
-    return versions.filter(function (val) {
-        return /v[0-9]+\.[0-9]+\.[0-9]+$/i.test(val);
+/**
+ *
+ * @param {String[]} versions - Schema Versions
+ */
+function getStableVersions (versions) {
+    return versions.filter((val) => {
+        return (/v[0-9]+\.[0-9]+\.[0-9]+$/i).test(val);
```
Comment (kunagpal) [inline scripts/build-docs.js]: `\d`

File: scripts/build-docs.js [hunk 4]
```diff
@@ -19,118 +19,157 @@ var path = require('path'),
...
+    console.info('Creating a Table of Contents...');
+
+    allversions = _.map(versions, (version) => {
+        let raw = version,
             semVersion = semver.SemVer(version),
             status,
             bootstrapClass;
+
         if (_.isArray(semVersion.prerelease) && semVersion.prerelease.length) {
             status = _.isArray(semVersion.prerelease) ? semVersion.prerelease[0] : semVersion.prerelease;
```
Comment (kunagpal) [inline scripts/build-docs.js]: Double `_.isArray` check.

**Review by kunagpal — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (codenirvana) [reply package.json:45]: `5.5.0` is the latest.

File: bin/compile.js [hunk 5]
```diff
@@ -1,41 +1,33 @@
...
-if (require.main == module) {
-    main();
+if (!program.schemaDir) {
+    console.error('\n  error: "--schema-dir" option is required');
+    process.exit(1);
 }
+
+let result = schema.compile(program.schema, program.schemaDir);
+
+fs.writeFileSync(program.output, JSON.stringify(result, null, 4));
```
Comment (czardoz) [inline bin/compile.js:31]: Can you change the indentation here to `2` since that's the value in `.editorconfig`?

File: bin/validate.js [hunk 6]
```diff
@@ -1,39 +1,28 @@
...
+    process.exit(1);
+}
+if (!program.schemaDir) {
+    console.error('\n  error: "--schema-dir" option is required');
+    process.exit(1);
 }
+
+schema.validate(program.input, program.schema, program.schemaDir) ?
+    console.info('Validation successful') :
+    console.error('Validation failed');
```
Comment (czardoz) [inline bin/validate.js:28]: The old code prints out the validation errors. Are we intentionally suppressing the output now?

File: bin/compile.js (hunk 5 above)
Comment (kunagpal) [reply bin/compile.js:31]: @czardoz We've switched to `4`.

File: .editorconfig [hunk 7]
```diff
@@ -10,5 +10,5 @@ max_length = 120
 trim_trailing_whitespace = true
 insert_final_newline = true
 
-[*.{json, yml, html, mustache, css}]
+[*.{json,yml,html,mustache,css}]
 indent_size = 2
```
Comment (czardoz) [inline .editorconfig:14]: @kunagpal Do you mean this should actually be `4`?

File: bin/compile.js (hunk 5 above)
Comment (czardoz) [reply bin/compile.js:31]: @kunagpal Am I missing something? https://github.com/postmanlabs/schemas/pull/20#discussion_r215491846

File: .editorconfig (hunk 7 above)
Comment (kunagpal) [reply .editorconfig:14]: My bad, I misread your previous comment. The indentation for inbuilt JSON files is 2, but 4 for generated JSON files (like the one above)

File: bin/validate.js (hunk 6 above)
Comment (codenirvana) [reply bin/validate.js:28]: @czardoz validation errors are printed out from `schema.validate` method, the return value is a boolean.

File: scripts/build-docs.js [hunk 8]
```diff
@@ -130,12 +130,8 @@ function buildToc (versions) {
             status,
             bootstrapClass;
 
-        if (_.isArray(semVersion.prerelease) && semVersion.prerelease.length) {
-            status = _.isArray(semVersion.prerelease) ? semVersion.prerelease[0] : semVersion.prerelease;
-        }
-        else {
-            status = 'stable';
-        }
+        status = Array.isArray(semVersion.prerelease) && semVersion.prerelease.length ?
```
Comment (kunagpal) [inline scripts/build-docs.js]: Lodash.

**Review by kunagpal — CHANGES_REQUESTED** [review]

**Review by kunagpal — APPROVED** [review]
